x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -194,5 +203,198 @@ export function MachineLearningStackManagementJobsProvider(
+        jobType === 'anomaly-detector' ? 'anomaly detection' : 'data frame analytics';
+
+      const results = title.match(
+        /(\d) (anomaly detection|data frame analytics) job[s]? read from file$/
+      );
+      expect(results).to.not.eql(null, `Expected regex results to not be null`);
+      const foundCount = results![1];
+      const foundJobTypeString = results![2];
+      expect(foundCount).to.eql(
+        jobCount,
+        `Expected job count to be '${jobCount}' (got '${foundCount}')`
+      );
+      expect(foundJobTypeString).to.eql(
+        jobTypeString,
+        `Expected job count to be '${jobTypeString}' (got '${foundJobTypeString}')`
+      );
+    },
+
+    async assertJobIdsSkipped(expectedJobIds: string[]) {
+      const subj = await testSubjects.find('mlJobMgmtImportJobsCannotBeImportedCallout');
+      const skippedJobTitles = await subj.findAllByTagName('h5');
+      const actualJobIds = (
+        await Promise.all(skippedJobTitles.map((i) => i.parseDomContent()))
+      ).map((t) => t.html());
+
+      expect(actualJobIds.sort()).to.eql(
+        expectedJobIds.sort(),
+        `Expected job ids to be '${JSON.stringify(expectedJobIds)}' (got '${JSON.stringify(
+          actualJobIds
+        )}')`
+      );
+    },
+
+    async importJobs() {
+      await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout', { timeout: 60 * 1000 });
+    },
+
+    async assertReadErrorCalloutExists() {
+      await testSubjects.existOrFail('~mlJobMgmtImportJobsFileReadErrorCallout');
+    },
+
+    async selectExportJobType(jobType: JobType) {
+      if (jobType === 'anomaly-detector') {
+        await testSubjects.click('mlJobMgmtExportJobsADTab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsADJobList');
+      } else {
+        await testSubjects.click('mlJobMgmtExportJobsDFATab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsDFAJobList');
+      }
+    },
+
+    async selectExportJobSelectAll() {
+      await testSubjects.click('mlJobMgmtExportJobsSelectAllButton');
+    },
+
+    async getDownload(filePath: string) {
+      return retry.tryForTime(5000, async () => {
+        expect(fs.existsSync(filePath)).to.be(true);
+        return fs.readFileSync(filePath).toString();
+      });
+    },
+
+    getExportedFile(fileName: string) {
+      return path.resolve(REPO_ROOT, `target/functional-tests/downloads/${fileName}.json`);
+    },
+
+    deleteExportedFiles(fileNames: string[]) {
+      fileNames.forEach((file) => {
+        try {
+          fs.unlinkSync(this.getExportedFile(file));
+        } catch (e) {
+          // it might not have been there to begin with
+        }
+      });
+    },
+
+    async selectExportJobs() {
+      await testSubjects.click('mlJobMgmtExportExportButton');
+    },
+
+    async assertExportedADJobsAreCorrect(expectedJobs: Array<{ job: Job; datafeed: Datafeed }>) {
```

Comment:
We're currently validating:
- The number of exported jobs is correct
- The exported jobs have the correct ids
- The exported jobs have the correct number of detectors
- The first detector of each exported job has the correct function
- 
(and something similar for data frame analytics jobs)

I'm wondering if we'd like to validate other/more details as well so we can be sure that all configuration items are actually exported. I guess comparing the whole job config is a bit of a pain due to nested objects and ordering of items?

Reply:
Yes, I deliberately didn't compare the whole job as this would mean this test would need to be updated every time a new property is added to the job.
We could compare more items, the real test here IMO is that something resembling a job is written to disk.
Under the hood this uses the standard AD GET requests with `exclude_generated` set. This isn't a test that that endpoint works correctly as elasticsearch has tests for that.

I meant to add a check for the datafeed ID too, I'll add that now

Reply:
I also thought about maybe importing the exported file again (delete jobs first) and see if the import can create the jobs - then we could be sure that at least valid job configurations have been exported (still not checking each detail, but as you pointed out: that would generate additional maintenance effort). As a downside that would rely on the import functionality to work - and if we have a bug there, we'd get a failure in `export` as well.
What do you think?

Reply:
Yeah that was my original plan for these tests, but went down this route as I was unsure how to write the export tests, so started the import ones first.
The real advantage of that approach would be to test the warnings for existing jobs, and then have to rename them. Which i've not covered in these tests.

Reply:
Alright, that sounds like something for a follow-up.